src/main/java/org/apache/xml/security/utils/DOMNamespaceContext.java

```diff
@@ -18,58 +18,124 @@
  */
 package org.apache.xml.security.utils;
 
-import java.util.HashMap;
-import java.util.Iterator;
-import java.util.Map;
-import java.util.Map.Entry;
+import org.w3c.dom.Node;
 
 import javax.xml.namespace.NamespaceContext;
+import java.util.Iterator;
+import java.util.Objects;
 
-import org.w3c.dom.Attr;
-import org.w3c.dom.Element;
-import org.w3c.dom.NamedNodeMap;
-import org.w3c.dom.Node;
+import static javax.xml.XMLConstants.DEFAULT_NS_PREFIX;
+import static javax.xml.XMLConstants.NULL_NS_URI;
+import static javax.xml.XMLConstants.XMLNS_ATTRIBUTE;
+import static javax.xml.XMLConstants.XMLNS_ATTRIBUTE_NS_URI;
+import static javax.xml.XMLConstants.XML_NS_PREFIX;
+import static javax.xml.XMLConstants.XML_NS_URI;
 
 /**
+ * This class adapts the {@link Node} namespace/prefix lookup API to that of {@link NamespaceContext}.
+ * There are some differences:
+ * <table>
+ *     <tr>
+ *         <th>Function</th>
+ *         <th>{@link NamespaceContext} API</th>
+ *         <th>{@link Node} API</th>
+ *     </tr>
+ *     <tr>
+ *         <td>Look up the prefix for a given namespace URI.</td>
+ *         <td>{@link NamespaceContext#getPrefix(String)}</td>
+ *         <td>{@link Node#lookupPrefix(String)}</td>
+ *     </tr>
+ *     <tr>
+ *         <td>Look up all the prefixes for a given namespace URI.</td>
+ *         <td>{@link NamespaceContext#getPrefixes(String)}</td>
+ *         <td>/</td>
+ *     </tr>
+ *     <tr>
+ *         <td>Look up the namespace URI for a given prefix.</td>
+ *         <td>{@link NamespaceContext#getNamespaceURI(String)}</td>
+ *         <td>{@link Node#lookupNamespaceURI(String)}</td>
+ *     </tr>
+ *     <tr>
+ *         <td>The default prefix.</td>
+ *         <td>{@link javax.xml.XMLConstants#DEFAULT_NS_PREFIX}</td>
+ *         <td>{@code null}</td>
+ *     </tr>
+ *     <tr>
+ *         <td>The default namespace URI.</td>
+ *         <td>{@link javax.xml.XMLConstants#NULL_NS_URI}</td>
+ *         <td>{@code null}</td>
+ *     </tr>
+ * </table>
  */
 public class DOMNamespaceContext implements NamespaceContext {
 
-    private Map<String, String> namespaceMap = new HashMap<>();
+    private Node context;
 
-    public DOMNamespaceContext(Node contextNode) {
-        addNamespaces(contextNode);
+    public DOMNamespaceContext(Node context) {
+        setContext(context);
     }
 
-    public String getNamespaceURI(String arg0) {
-        return namespaceMap.get(arg0);
+    public void setContext(Node context) {
```

Comment:
Is this only used for testing? It would be better to remove it and make context final.

Reply:
It's not used here yet, but it's going to be used in SANTUARIO-532, which is what I was referring to in my comment on the JIRA issue SANTUARIO-534:

> 3. Added DOMNamespaceContext.setContext(Node context) to allow reusing a single instance with multiple Node contexts, in anticipation of using it in SANTUARIO-532.

If you insist I can remove it here, but I was hoping to keep it in anticipation of a future PR...

Reply:
OK if it's needed for the other PR then it's fine to leave it.